pkg/cmd/dev/ui.go

```diff
@@ -547,6 +552,85 @@ func makeUIStorybookCmd(d *dev) *cobra.Command {
 	return storybookCmd
 }
 
+// makeUICrdbApiClientCmd initializes the 'ui crdb-api-client' subcommand. which
+// generates JS protobuf client package and hoists it to source tree.
+func makeUICrdbApiClientCmd(d *dev) *cobra.Command {
+	return &cobra.Command{
+		Use:   "crdb-api-client",
+		Short: "Build crdb-api-client library",
+		Long:  ``,
+		Args:  cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, commandLine []string) error {
+			// Create a context that cancels when OS signals come in.
+			ctx, stop := signal.NotifyContext(d.cli.Context(), os.Interrupt, os.Kill)
+			defer stop()
+			bazelBin, err := d.getBazelBin(d.cli.Context(), []string{})
+			if err != nil {
+				return err
+			}
+			dstDirs, err := getUIDirs(d)
+			if err != nil {
+				return err
+			}
+
+			// Ensure node dependencies are up-to-date.
+			err = d.exec.CommandContextInheritingStdStreams(
+				ctx,
+				"pnpm",
+				"--dir",
+				dstDirs.root,
+				"install",
+			)
+			if err != nil {
+				log.Fatalf("failed to fetch node dependencies: %v", err)
+			}
+
+			// Fetch all node dependencies (through Bazel) to ensure things are up-to-date
+			err = d.exec.CommandContextInheritingStdStreams(
+				ctx,
+				"bazel",
+				"fetch",
```

Comment:
Why is the fetching necessary? Does `build` not fetch new dependencies by itself?

Reply:
Agree, it was unnecessary. Removed.

Reply:
> Tests failing

Done